CLAUDE.md

```diff
@@ -102,6 +102,274 @@ When setting array attributes, the schema uses specific patterns:
 /secrets/Hetzner Api Token: {$source: {component: "credential-id", path: "/secrets/Hetzner::Credential::ApiToken"}}
 ```
 
+#### Microsoft Azure Components
+
+System Initiative supports Microsoft Azure infrastructure management using Azure Resource Manager (ARM) resource types.
+
+#### IMPORTANT NOTE Regarding Azure/Microsoft
+
+Users may use the terms "Azure" and "Microsoft" interchangibly to refer to Microsoft Azure components, schemas, resources, etc. Respond to users accordingly.
+
+When looking up a schema by its name, the schema name will start with Microsoft, not Azure.
+
+##### Key Microsoft Azure Schemas
+
+- **Microsoft Credential** - Authentication credentials for Azure - the secret details must be provided by the user
+- **Microsoft.Resources/locations** - Azure region specification (e.g., "eastus", "westus2") - prefer using this component to just using a location string
+- **Microsoft.Resources/subscription** - Azure subscription ID reference - prefer using this component to just using a subscription ID string
+- **Microsoft.Resources/resourceGroups** - Container for related Azure resources (required for most resources) - resource groups can be managed via CRUD by System Initiative
+
+When creating any Microsoft Azure component that is not a Microsoft Credential, Microsoft.Resources/locations, or Microsoft.Resources/subscription those three components must exist to be subscribed to by the component. The only exception to this is the Microsoft.Resources/resourceGroups component, which only requires a Microsoft Credential and Microsoft.Resources/locations.
+
+If a resource group does not exist, create one for the user.
```

Comment:
I am not entirely sure how granularly people use resource groups. I wonder if we should prompt for create or import?